### PR TITLE
Added the domains module

### PR DESCRIPTION
**Pull request by johnake**
### Context

This configuration deploys the Azure Front - which is a Content Delivery Network (CDN) that provides fast, reliable and secure access between user and Applications static and dynamic web contents.

### Changes proposed in this pull request

Added a 'domains' folder in the terraform directory that contains the configuration for the change above.

### Guidance to review

Include any useful information needed to review this change.
Include any dependencies that are required for this change.

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by malcolmbaig** [comment]
Hi @johnake — could you fill out the PR description a bit? Nothing too heavy needed, just a basic description of what this is and why we're adding it.

File: terraform/domains/infrastructure/config/zones.tfvars.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  "tags": {
+    "Parent Business": "Teacher Training and Qualifications",
+    "Portfolio": "Early Years and Schools Group",
+    "Product": "Database of Qualified Teachers",
+    "Service": "Teacher Training and Qualifications",
+    "Service Line": "Teaching Workforce",
+    "Service Offering": "Database of Qualified Teachers",
+    "Environment": "Prod"
+  },
+  "deploy_default_records": false
```
Comment (nickdevdfe) [inline terraform/domains/infrastructure/config/zones.tfvars.json:26]: Service Offering and Product should be  'Find a Lost TRN'

File: terraform/domains/environment_domains/variables.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
+variable "zone" {
+  type        = string
+  description = "Name of DNS zone"
+  default     = "find-a-lost-trn-in-england.education.gov.uk"
```
Comment (nickdevdfe) [inline terraform/domains/environment_domains/variables.tf:4]: should be 'find-a-lost-trn.education.gov.uk'

File: terraform/domains/infrastructure/config/zones.tfvars.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  "hosted_zone": {
+    "find-a-lost-trn-in-england.education.gov.uk": {
```
Comment (nickdevdfe) [inline terraform/domains/infrastructure/config/zones.tfvars.json:3]: should be 'find-a-lost-trn.education.gov.uk'

**Review by nickdevdfe — APPROVED** [review]
Tested, reviewed and approved

**Comment by nickdevdfe** [comment]
[Deploy AKS infra resources](https://trello.com/c/4mkzwhhw/545-deploy-aks-infra-resources)

**Pull request merged**